utilities/src/bytes.rs

```diff
@@ -28,6 +29,21 @@ use serde::{
 };
 use smol_str::SmolStr;
 
+thread_local! {
+    static UNCHECKED_DESERIALIZE: Cell<bool> = const { Cell::new(false) };
```

Comment:
@ljedrz any final comments on this approach?

Reply:
should be a fine workaround for now; in the future, once we move to `bincode 2`, we might be able to utilize something like [decode_from_slice_with_context](https://docs.rs/bincode/latest/bincode/fn.decode_from_slice_with_context.html) instead